### PR TITLE
Clean-up some broken code in the test framework

### PR DESCRIPTION
**Pull request by clokep**
I think this was left-over cruft from switching from a single datastore to multiple datastores.

### CODE REVIEW

**Comment by clokep** [comment]
The non-whitespace version of this diff is much more readable: https://github.com/matrix-org/synapse/pull/8514/files?w=1

**Review by richvdh — APPROVED** [review]
lgtm

File: tests/server.py [hunk 1]
```diff
@@ -367,8 +367,6 @@ def setup_test_homeserver(cleanup_func, *args, **kwargs):
     """
     server = _sth(cleanup_func, *args, **kwargs)
 
-    database = server.config.database.get_single_database()
```
Comment (clokep) [inline tests/server.py:370]: As far as I can tell this isn't doing anything special (like causing the database to get cached or something).

Comment (richvdh) [reply tests/server.py:370]: agreed. I came across it the other day too.

**Pull request merged**